selfservice/hook/session_issuer.go

```diff
@@ -52,6 +56,19 @@ func (e *SessionIssuer) executePostRegistrationPostPersistHook(w http.ResponseWr
 	}
 
 	if a.Type == flow.TypeAPI {
+		if ok, _ := e.r.SessionTokenExchangePersister().CodeExistsForFlow(r.Context(), a.ID); ok {
+			if err := e.r.SessionTokenExchangePersister().UpdateSessionOnExchanger(r.Context(), a.ID, s.ID); err != nil {
+				return errors.WithStack(err)
+			}
+			returnTo, err := x.SecureRedirectTo(r, e.r.Config().SelfServiceBrowserDefaultReturnTo(r.Context()), a.SecureRedirectToOpts(r.Context(), e.r)...)
```

Comment:
The idea here being that this would open the registered URL of the app? So for example `myapp://post-login`?

Reply:
Yes, exactly. The behaviour of a flow when a code was requested is that we always redirect to the `return_to` URL and then let the client fetch the session token (on success) or refetch the flow (on failure).